--- a/render/src/modules.ts	2026-09-30 14:29:13.670584192 +0200
+++ b/render/src/modules.ts	2026-09-30 14:29:13.684999427 +0200
@@ -33,16 +33,16 @@
 };
 
 // Expose react
-const react = resolveCjsModule(/\/react-(?!dom[-.])[^/]+\.js$/, (r) => typeof r.useState === "function" && typeof r.useEffect === "function");
+const react = resolveCjsModule(/\/(react-(?!dom[-.])[^/]+|framework-[^/]+)\.js$/, (r) => typeof r.useState === "function" && typeof r.useEffect === "function");
 if (react) { react.default ??= react; modules["react"] = react; }
 else { coreTrace.warn("modules", "Failed to resolve React module"); }
 
-const jsxRT = resolveCjsModule(/\/jsx-runtime-[^/]+\.js$/, (r) => typeof r.jsx === "function" && typeof r.jsxs === "function");
+const jsxRT = resolveCjsModule(/\/(jsx-runtime|framework)-[^/]+\.js$/, (r) => typeof r.jsx === "function" && typeof r.jsxs === "function");
 if (jsxRT) { jsxRT.default ??= jsxRT; modules["react/jsx-runtime"] = jsxRT; }
 else { coreTrace.warn("modules", "Failed to resolve react/jsx-runtime module"); }
 
 // Tidal tree shakes hydrateRoot
-const reactDomPath = /\/react-dom(-client)?-[^/]+\.js$/;
+const reactDomPath = /\/(react-dom(-client)?|framework)-[^/]+\.js$/;
 const reactDom = resolveCjsModule(reactDomPath, (r) => typeof r.createRoot === "function");
 
 // Fallback for react-dom/client
